File: frontend/src/api/coach.ts
import api from "./client";
import type { ChatRequest, ChatResponse, InsightsResponse } from "@/types";

export const coachApi = {
  chat: (data: ChatRequest) =>
    api.post<ChatResponse>("/coach/chat", data).then((r) => r.data),

  insights: () =>
    api.get<InsightsResponse>("/coach/insights").then((r) => r.data),

  dailyNutritionSummary: (date: string) =>
    api.get<InsightsResponse>("/coach/insights", { params: { date } }).then((r) => r.data),
};

File: backend/app/api/v1/coach.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.deps import get_current_user, get_db
from app.models.user import User
from app.schemas.ai import ChatRequest, ChatResponse, InsightsResponse
from app.services.coach_service import chat, generate_insights

router = APIRouter(prefix="/coach", tags=["coach"])


@router.post("/chat", response_model=ChatResponse)
async def coach_chat(
    data: ChatRequest,
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """Chat with the AI health coach."""
    return await chat(db, current_user, data)


@router.get("/insights", response_model=InsightsResponse)
async def get_insights(
    db: AsyncSession = Depends(get_db),
    current_user: User = Depends(get_current_user),
):
    """Get AI-generated health insights based on recent data."""
    return await generate_insights(db, current_user)
